kornia/augmentation/augmentation.py

```diff
@@ -412,22 +412,22 @@ class RandomRotation(AugmentationBase):
     def __init__(self, degrees: FloatUnionType = 45.0, return_transform: bool = False) -> None:
         super(RandomRotation, self).__init__(F._apply_rotation, return_transform)
         self.degrees = degrees
-        self._params: Dict[str, torch.Tensor] = {}
 
     def __repr__(self) -> str:
         repr = f"(degrees={self.degrees}, return_transform={self.return_transform})"
         return self.__class__.__name__ + repr
 
-    def set_params(self, batch_size: int, degrees: FloatUnionType):
-        self._params = pg._random_rotation_gen(batch_size, degrees)
+    @staticmethod
+    def get_params(batch_size: int, degrees: FloatUnionType):
```

Comment:
I think we should discuss this change, and what should be the expected behaviour for the API. I would go an independent factory class that maybe generates random distributions. Besides, one use case I think it's interesting to cover is to generate same random distributions for let's say siamese like samples. Let's discuss a little bit this before proceeding.

Reply:
BTW, just noticed that albumentations add a random parameter in every single function, maybe we could same here. See: https://albumentations.readthedocs.io/en/latest/api/augmentations.html#albumentations.augmentations.transforms.Blur

Reply:
I think either way is good to go though I personally prefer to keep a function as stateless as it could. Also, I do not think using a ```_params``` variable will help much as a user. It will update the workflow from

```python
transform = SomeTransform(param_ranges)
params = SomeTransform.get_params(param_ranges)
ouput_a = transform(batch_size, params)(IMAGES_A)
ouput_b = transform(batch_size, params)(IMAGES_B)
```
to
```python
transform = SomeTransform(param_ranges)
ouput_a = transform(batch_size)(IMAGES_A)
params = SomeTransform._params
ouput_b = transform(batch_size, params)(IMAGES_B)
```

Reply:
I think what may help is to add a random seed on each call.

In such cases, we will have:
```python
transform = SomeTransform(param_ranges)
output_a = transform(params, random_seed=1)(IMAGES_A)
output_b = transform(params, random_seed=1)(IMAGES_B)
```
It looks more concise and compact. Well, for what I can think of,  the problem is how can we control the scope of those random_seeds. If I remember right, torch.set_manual_seed() will change the whole random state of torch framework.

Reply:
ok, then go ahead with the random_seed parameters. I would add one more, `p=0.5` for each of the transforms For the getter, the only requirement is to have a clean API for cases like siamese training when you might want to apply same params to a set of tensor.

Reply:
Sure, I can help with that. It should not have massive changes. Can you merge in this PR then I will start another one for that?

What do you mean of the parameter ```p=0.5```? What is that for? I did not get it. e.g. why do we need p=0.5 for colorjitter?

Reply:
to have an homogeneous API I guess. ColorJitter is a bad API example, it's just a combination of 4 color transforms, and why this 4 ? -> class inherited from torchvision but not sure the reason of its existence at all. In terms of data augmentation why not having by default a mask that decided whether to apply the transform (color o geometric).

Reply:
I'll merge this right now